Add DSM 7 support to synology.md

TallDragon opened the pull request:
Added the installation for DSM 7. Now a Synology specific docker is needed.

wborn commented on installation/synology.md:
```diff
@@ -3,10 +3,113 @@ layout: documentation
 
-The [DiskStation by Synology](https://www.synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages.
-We are proud to be able to provide an [openHAB Synology package](https://github.com/openhab/openhab-syno-spk).
+The [DiskStation by Synology](https://www.synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages. For versions of DSM before 7.0 please refer to the second half of this page.
```
Each sentence in Markdown should be placed on a new line so it easier to review and compare changes.

```suggestion
The [DiskStation by Synology](https://www.synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages.
For versions of DSM before 7.0 please refer to the second half of this page.
```

TallDragon replied:
Tnx for the review, very useful feedback.

wborn commented on installation/synology.md:
```diff
@@ -3,10 +3,113 @@ layout: documentation
+
+## Prerequisite - prepare the Synology OS for openhab
+
+Synology NAS is based on unix but it has limitations since it is not a full server. There are some shortcuts you have to take to make it work. For the use of openHAB you need some preparation steps. For this create a ssh session into your NAS. For reference how to do that: I found this [youtube tutorial](https://www.youtube.com/watch?v=BCCIMRbAUp8).
```
My Synlogy NAS has a Linux kernel.

```suggestion
Synology NAS is based on Linux but it has limitations since it is not a full server.
There are some shortcuts you have to take to make it work.
For the use of openHAB you need some preparation steps.
For this create a ssh session into your NAS.
For reference how to do that: I found this [youtube tutorial](https://www.youtube.com/watch?v=BCCIMRbAUp8).
```

wborn commented on installation/synology.md:
````diff
@@ -3,10 +3,113 @@ layout: documentation
+sudo synouser --add openhab habopen
+sudo synogroup --add openhab openhab
+```
+If you are going to use an usb like a z-wave dongle add another group for openhab and change the port access for ttyACM0 if that is the port for your usb device
````
```suggestion
If you are going to use an usb like a Z-Wave dongle add another group for openHAB and change the port access for ttyACM0 if that is the port for your USB device
```

Please also use the proper capitalization for brands/abbreviations/products in the other parts:

* Docker
* openHAB
* Windows

I created #1671 to correct similar issues changes recently. :wink:

TallDragon replied:
Ok. clear

wborn reviewed (commented):
Thanks for adding all the installation steps. :+1: 

I have a few comments below :slightly_smiling_face:

TallDragon commented:
As advised in the first review I changed the sentences per line number, changed the brand names to correct case and even repaired the original content to these feedback requests. 
Please let me know what is needed more.

wborn commented:
> As advised in the first review I changed the sentences per line number, changed the brand names to correct case and even repaired the original content to these feedback requests.

Maybe something went wrong? Because when I review the [files](https://github.com/openhab/openhab-docs/pull/1673/files) I still see that there are many sentences not starting on new lines and brand names without proper capitalization?

TallDragon commented:
File checked in, giving up on DCO. No clue how that works, but I suck at it and will not spent any time in this further. Feel free to throw my work away if this DCO is mandatory. 
Completely frustrated. If someone can spent some time to help people like me out in how to help here I would be greatful. But don't feel motivated to continue anymore.

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
+The [DiskStation by Synology](https://www.synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages. 
+For versions of DSM before 7.0 please refer to the second half of this page.
+
+## Prerequisite - prepare the Synology OS for OpenHAB
```
```suggestion
## Prerequisite - prepare the Synology OS for openHAB
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
+
+Synology NAS is based on Linux but it has limitations since it is not a full server.
+There are some shortcuts you have to take to make it work.  
+For the use of OpenHAB you need some preparation steps.
```
```suggestion
For the use of openHAB you need some preparation steps.
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
+
+## Docker
+
+DSM 7 does not longer support java8 directly and therefor OpenHAB has to run in a Docker container. 
```
```suggestion
DSM 7 does no longer support Java 11 directly and therefor openHAB has to run in a Docker container. 
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
+
+DSM 7 does not longer support java8 directly and therefor OpenHAB has to run in a Docker container. 
+And that also makes the installation easier to maintain. 
+It works out of the box a bit different then the normal Docker installation as described in de OpenHAB documententation.
```
```suggestion
It works out of the box a bit different then the normal Docker installation as described in de openHAB documententation.
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
+- Search for "Docker" and install.
+- Click "Open".
+- Go to "Registry", here are all the available Docker images and search for openhab.
+- Choose the "openhab/openhab" image and click "Download". Select "latest" since that is the latest stable version of OpenHAB. The Docker image is added to "Image"
```
```suggestion
- Choose the "openhab/openhab" image and click "Download". Select "latest" since that is the latest stable version of openHAB. The Docker image is added to "Image"
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
+
+### Summary 
+
+Here you can check if all changes where set correctly for your OpenHAB container, enable "Run this container after the wizard is finished" and press "Done".
```
```suggestion
Here you can check if all changes where set correctly for your openHAB container, enable "Run this container after the wizard is finished" and press "Done".
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
+### Summary 
+
+Here you can check if all changes where set correctly for your OpenHAB container, enable "Run this container after the wizard is finished" and press "Done".
+OpenHAB is now running on your NAS. 
```
```suggestion
openHAB is now running on your NAS. 
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
+
+If you select the container running and press "Details" you will end up on  a windows where you get more information on this running container. 
+One particular menu item you have to check is "Terminal". 
+In here you see the name of the container left from the black field and if you press that you'll get the Karaf terminal for OpenHAB.
```
```suggestion
In here you see the name of the container left from the black field and if you press that you'll get the Karaf terminal for openHAB.
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
+
+If you click on "Create' it creates a bash terminal, click on bash and you'll have access to a shell in the container.
+
+By now the container is up and running, goto "http://[your-NAS-ip-or-host-name]:8080" and the welcome to OpenHAB.
```
```suggestion
By now the container is up and running, goto "http://[your-NAS-ip-or-host-name]:8080" and the welcome to openHAB.
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
+If you click on "Create' it creates a bash terminal, click on bash and you'll have access to a shell in the container.
+
+By now the container is up and running, goto "http://[your-NAS-ip-or-host-name]:8080" and the welcome to OpenHAB.
+If you use p.e. a Z-Wave dongle on ttyACM0 you can now add the Z-Wave binding. Once OpenHAB has been configured under the binding Z-Wave add a controller manually and change the port to /dev/ttyACM0. The Z-Wave controller gets online and you will discover the Z-Wave devices. If this is not working, please verify you enabled high privilege.
```
```suggestion
If you use p.e. a Z-Wave dongle on ttyACM0 you can now add the Z-Wave binding. Once openHAB has been configured under the binding Z-Wave add a controller manually and change the port to /dev/ttyACM0. The Z-Wave controller gets online and you will discover the Z-Wave devices. If this is not working, please verify you enabled high privilege.
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
 
-The [DiskStation by Synology](https://www.synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages.
-We are proud to be able to provide an [openHAB Synology package](https://github.com/openhab/openhab-syno-spk).
+The [DiskStation by Synology](https://www.Synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages. 
```
```suggestion
The [DiskStation by Synology](https://www.synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages.
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
-The [DiskStation by Synology](https://www.synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages.
-We are proud to be able to provide an [openHAB Synology package](https://github.com/openhab/openhab-syno-spk).
+The [DiskStation by Synology](https://www.Synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages. 
+We are proud to be able to provide an [OpenHAB Synology package](https://github.com/openhab/openhab-syno-spk).
```
```suggestion
We are proud to be able to provide an [openHAB Synology package](https://github.com/openhab/openhab-syno-spk).
```

wborn commented on installation/synology.md:
```diff
@@ -3,14 +3,133 @@ layout: documentation
 ## Prerequisite - Install Java
 
-Current releases of openHAB require Oracle Java due to the generation of certificates based on [EC](https://en.wikipedia.org/wiki/Elliptic_curve_cryptography) which is not available in free Java implementations.
+Current releases of OpenHAB require Oracle Java due to the generation of certificates based on [EC](https://en.wikipedia.org/wiki/Elliptic_curve_cryptography) which is not available in free Java implementations.
```
```suggestion
Current releases of openHAB require Oracle Java due to the generation of certificates based on [EC](https://en.wikipedia.org/wiki/Elliptic_curve_cryptography) which is not available in free Java implementations.
```

wborn commented on installation/synology.md:
```diff
@@ -23,9 +142,9 @@ To update Oracle JDK to Java8, click the Install Java / Upgrade Java button and
 
 **Armada 370/XP** systems running DSM 6 require hard float ABI rather than the soft float used on DSM 5.
-If there is not a Java8 package in the Package Center, your machine may not be supported by the [Synology package](https://www.synology.com/en-us/dsm/packages/Java8).
+If there is not a Java8 package in the Package Center, your machine may not be supported by the [Synology package](https://www.Synology.com/en-us/dsm/packages/Java8).
```
```suggestion
If there is not a Java8 package in the Package Center, your machine may not be supported by the [Synology package](https://www.synology.com/en-us/dsm/packages/Java8).
```

wborn commented on installation/synology.md:
```diff
@@ -23,9 +142,9 @@ To update Oracle JDK to Java8, click the Install Java / Upgrade Java button and
+If there is not a Java8 package in the Package Center, your machine may not be supported by the [Synology package](https://www.Synology.com/en-us/dsm/packages/Java8).
 
-The best option for older unsupported Synology models such as **PowerPC** and **ARMv5** is to include the [PC-Loadletter](https://pcloadletter.co.uk/2011/08/23/java-package-for-synology/) Repo.
+The best option for older unsupported Synology models such as **PowerPC** and **ARMv5** is to include the [PC-Loadletter](https://pcloadletter.co.uk/2011/08/23/java-package-for-Synology/) Repo.
```
```suggestion
The best option for older unsupported Synology models such as **PowerPC** and **ARMv5** is to include the [PC-Loadletter](https://pcloadletter.co.uk/2011/08/23/java-package-for-synology/) Repo.
```

wborn commented on installation/synology.md:
```diff
@@ -36,7 +155,7 @@ The **Java8 PowerPC** versions are available from [Oracle](https://www.oracle.co
 
 Download the latest SPK package: [Releases](https://github.com/openhab/openhab-syno-spk/releases)
-The SPK is a wrapper to download the latest openHAB release and does not contain openHAB itself.
+The SPK is a wrapper to download the latest OpenHAB release and does not contain OpenHAB itself.
```
```suggestion
The SPK is a wrapper to download the latest openHAB release and does not contain openHAB itself.
```

wborn commented on installation/synology.md:
```diff
@@ -50,7 +169,7 @@ If your NAS cannot connect to the internet, the installer will tell you to downl
 ## Configuration
 
-The location of configuration files by openHAB differs depending on wether or not a shared folder named `public` exists:
+The location of configuration files by OpenHAB differs depending on wether or not a shared folder named `public` exists:
```
```suggestion
The location of configuration files by openHAB differs depending on wether or not a shared folder named `public` exists:
```

wborn commented on installation/synology.md:
```diff
@@ -59,15 +178,16 @@ For certain Synology models the `public` folder is created automatically during
 ## Logging
 
-The openHAB log files can be found here:
+The OpenHAB log files can be found here:
```
```suggestion
The openHAB log files can be found here:
```

wborn commented on installation/synology.md:
```diff
@@ -59,15 +178,16 @@ For certain Synology models the `public` folder is created automatically during
 
 Synology does not document which models will automatically create a public folder.
-Refer to the Synology knowlegde base article on [Shared Folder](https://www.synology.com/en-us/knowledgebase/DSM/help/DSM/AdminCenter/file_share_desc) to learn more.
+Refer to the Synology knowlegde base article on [Shared Folder](https://www.Synology.com/en-us/knowledgebase/DSM/help/DSM/AdminCenter/file_share_desc) to learn more.
```
```suggestion
Refer to the Synology knowlegde base article on [Shared Folder](https://www.synology.com/en-us/knowledgebase/DSM/help/DSM/AdminCenter/file_share_desc) to learn more.
```

wborn reviewed (commented):
Thanks for processing my comments!

You've already signed off your commits, so when this pull request is merged the proper sign off can be added by the maintainers. So the failing DCO check can be ignored. 

I think a bit too much words were captilalized now. So if you smash the buttons to merge my suggestions I think all will be OK afterwards. :slightly_smiling_face:

wborn commented:
OK almost there. Please also hit the button on the comments that are hidden after clicking "Load more...":

![Screenshot from 2021-11-14 22-02-53](https://user-images.githubusercontent.com/12213581/141698556-06381767-a231-4839-9ec4-624175139735.png)

wborn reviewed (approved):
Thank you very much! It looks good to me now. :+1: 
Now we have to wait a bit for a docs maintainer to have a second look and merge it. :hourglass_flowing_sand:

TallDragon commented:
tnx for helping out. Still puzzled about DCO. Spent over 4 hours on it and it still is generating all kinds of errors. 
I'll continue on my openHAB (with lower case o) and maybe step in on some programming or other documentation work.

wborn commented:
> Still puzzled about DCO. Spent over 4 hours on it and it still is generating all kinds of errors.

The issue is that the DCO check does not work well with GitHub suggestions.
The DCO check expects every commit to end with:

`Signed-off-by: name <email>`

However when you accept suggestions GitHub adds the followign at the end of commits:

`Co-authored-by: name <email>`

The DCO check can also be set to pass by maintainers after clicking a button :wink: 

![Screenshot from 2021-11-14 22-24-17](https://user-images.githubusercontent.com/12213581/141699194-1f9fcd37-1abb-4e6e-ac19-11b2d213ea8f.png)

wborn commented:
Looks like you've just undone all the suggestions with your latest merge. :thinking:

TallDragon commented:
That problem is the result of me getting lost in DCO en github gui correcting by remerging. 
I'm trying again to get this solved. More drama comming up.

TallDragon commented:
Oke, please help. I have the file with all the changes on my pc. I have been using git for longer, but this is the first time I'm lost.
So how do I get the corrected file committed in this project for review. pulling/pushing/rebasing/forcing.... Panic.

wborn commented:
You can also do it in the GitHub UI by going to [Files](https://github.com/openhab/openhab-docs/pull/1673/files) and choose "Edit file"

![Screenshot from 2021-11-14 22-44-19](https://user-images.githubusercontent.com/12213581/141699820-881ba386-6c42-416c-9086-00e92e049f59.png)

Just don't touch it again aftwards. :wink:

TallDragon commented:
Please check if it checked in... And it is the version with the openHAB brand.

wborn commented:
Yes it seems to be fixed again! :slightly_smiling_face:

TallDragon commented:
Thanks for validating. I will now touch nothing anymore till this is merged in main and I can safely cleanup everything

Confectrician commented:
Hey @TallDragon 

Sorry for the delay and thanks for the contribution.
No worries about the DCO Check.
I will take care of everything and will inform you if there are any steps left from your side.

Also thanks to @wborn for reviewing and helping out here.
I will ahve a look and check if we are ready to merge. :)

Confectrician reviewed (approved):
Thanks 👍🏼
